File: backend/src/main/java/com/example/backend/service/NotificationService.java
package com.example.backend.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.example.backend.model.respon.NotificationResponse;

public interface NotificationService {
    // lấy danh sách thông báo
    Page<NotificationResponse> getNotifications(Long userId, Pageable pageable);

    // đánh dấu 1 thông báo là đã đọc
    Boolean markNotificationAsRead(Long userId, Long notificationId);

    // đánh dấu tất cả thông báo là đã đọc
    Boolean markAllNotificationsAsRead(Long userId);
}

File: frontend/src/service/impl/NotificationService.ts
import { http } from "../http/http";
import type { ApiDataResponse, Page, NotificationResponse } from "../../utils/ResponType";

export const getNotifications = (currentUserId: number, page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<NotificationResponse>>>(
        `/notifications?currentUserId=${currentUserId}&page=${page}&size=${size}`
    );
};

export const markNotificationAsRead = (currentUserId: number, notificationId: number) => {
    return http.put<ApiDataResponse<boolean>>(`/notifications/${notificationId}/read?currentUserId=${currentUserId}`);
};

export const markAllNotificationsAsRead = (currentUserId: number) => {
    return http.put<ApiDataResponse<boolean>>(`/notifications/read-all?currentUserId=${currentUserId}`);
};
